libra_toolbox/neutron_detection/activation_foils/foils.py

```diff
@@ -0,0 +1,198 @@
+                 efficiency_bounds=None,
+                 geometric_efficiency=1,
+                 photon_counts=None,
+                 photon_counts_uncertainty=None,
+                 tzinfo=ZoneInfo('America/New_York')):
+        # Ensure times are all datetime objects with timezones
+        self.time_generator_off = convert_to_datetime(time_generator_off)
+        self.start_time_counting = convert_to_datetime(start_time_counting)
+        self.distance_from_center_of_target_plane = distance_from_center_of_target_plane
+        self.real_count_time = real_count_time
+        if live_count_time:
+            self.live_count_time = live_count_time
+        else:
+            self.live_count_time = real_count_time
+        self.name = name
+        self.generator = generator
+        self.run = run
+
+        self.total_eff_coeff = total_eff_coeff
+        self.intrinsic_eff_coeff = intrinsic_eff_coeff
+        self.geometric_efficiency = geometric_efficiency
+        self.efficiency_bounds = efficiency_bounds
+        self.photon_counts = photon_counts
+        self.photon_counts_uncertainty = photon_counts_uncertainty
+
+    
+
+
+
+class Foil:
+    def __init__(self, mass: float, thickness: float, name=''):
+        """
+        Base class for neutron activation foils.
+        :param mass: Mass of the foil in grams
+        :param thickness: Thickness of the foil in cm
+        """
+        self.mass = mass
+        self.thickness = thickness
+        self.name = name
+        self.half_life = None
+        self.decay_constant = None
+        self.reaction = None
+        self.branching_ratio = None
+        self.photon_energies = None
+        self.cross_section = None  # in barns (1 barn = 1e-24 cm^2)
+        self.density = None  # in g/cm^3
+        self.atomic_mass = None
+        self.abundance = None
+        self.mass_attenuation_coefficient = None  # in cm^2/g
+    
+    def get_properties(self):
+        """Returns the properties of the foil."""
+        return {
+            "Name": self.name,
+            "Mass (g)": self.mass,
+            "Thickness (cm)": self.thickness,
+            "Decay Constant (1/s)": self.decay_constant,
+            "Reaction": self.reaction,
+            "Branching Ratio": self.branching_ratio,
+            "Photon Energies": self.photon_energies,
+            "Cross Section (barns)": self.cross_section,
+            "Density (g/cm³)": self.density,
+            "Atomic Mass": self.atomic_mass,
+            "Abundance": self.abundance,
+            "Mass Attenuation Coefficient (cm²/g)": self.mass_attenuation_coefficient
+        }
+    
+    def get_decay_constant(self):
+        self.decay_constant = np.log(2) / self.half_life.to(ureg.s)
+
+    def get_atoms(self):
+        self.atoms = (self.abundance 
+                      * (self.mass 
+                         * (6.022141e23 * ureg.particle / ureg.mol) 
+                         / self.atomic_mass))
```

Comment:
Similarly, this should be a property:

```suggestion
    @property
    def nb_atoms(self):
        return self.abundance  * self.mass / self.atomic_mass
```